linter/lints/rfc/lint_crl_no_empty_revoked_certificates_list.go

```diff
@@ -0,0 +1,60 @@
+package rfc
+
+/*
+ * ZLint Copyright 2023 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)
+
+type crlNoEmptyRevokedCertsList struct{}
+
+/************************************************
+RFC 5280: 5.1.2.6
+When there are no revoked certificates, the revoked certificates list MUST be
+absent.
+************************************************/
+
+func init() {
+	lint.RegisterRevocationListLint(&lint.RevocationListLint{
+		LintMetadata: lint.LintMetadata{
+			Name:          "e_crl_no_empty_revoked_certificates_list",
+			Description:   "When there are no revoked certificates, the revoked certificates list MUST be absent.",
+			Citation:      "RFC 5280: 5.1.2.6",
+			Source:        lint.RFC5280,
+			EffectiveDate: util.RFC5280Date,
+		},
+		Lint: NewCrlNoEmptyRevokedCertsList,
+	})
+}
+
+func NewCrlNoEmptyRevokedCertsList() lint.RevocationListLintInterface {
+	return &crlNoEmptyRevokedCertsList{}
+}
+
+func (l *crlNoEmptyRevokedCertsList) CheckApplies(c *x509.RevocationList) bool {
+	return true
+}
+
+func (l *crlNoEmptyRevokedCertsList) Execute(c *x509.RevocationList) *lint.LintResult {
+	if c.RevokedCertificates != nil && len(c.RevokedCertificates) == 0 {
```

Comment:
Something not relevant to this PR, but I noticed passing through: this suggests a semantic difference between `nil` and "non-`nil` but 0 len`, which is not documented in crl_x509.

Reply:
Good catch. I think the reason it's not documented is that to me, the person writing crl_x509, this was (in large part) _the whole point_. Being able to distinguish between "has no RevokedCertificates list" and "has an empty RevokedCertificates list" was one of the core things I wanted, and so it was just obvious to me that that's how it works, and so I completely forgot to document it.

All that said, this subtle distinction was (correctly, imo) rejected by the Go maintainers during the upstreaming process. So this lint will have to change when we unfork.